dev/reference: add some description about max_execution_time (#1430)
pingcap/docs: dev/reference/performance/optimizer-hints.md

```

This variable is used to remind the optimizer to use the `Hash Join` algorithm. This algorithm executes threads concurrently. It runs faster but takes up more memory.

> ### MAX\_EXECUTION\_TIME(N)

Reviewer: Please use the same format as v2.1.

> The `MAX_EXECUTION_TIME` hint is permitted only for SELECT statements. It places a limit N (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:

Reviewer 2: Would it be better this way?

```suggestion
The `MAX_EXECUTION_TIME` hint only applies to `SELECT` statements. It places a limit `N` (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:
```

```
MAX_EXECUTION_TIME(N)
```

Example with a timeout of 1 second (1000 milliseconds):

```sql
SELECT /*+ MAX_EXECUTION_TIME(1000) */ * FROM t1 INNER JOIN t2 WHERE ...
```

> In addition to the hint, the `max_execution_time` system variable also applies to the current statement.

Reviewer 3: ```suggestion
In addition to this hint, the `max_execution_time` system variable also limits the execution time of a statement.
```
